### PR TITLE
add file mode parameter to handle .toml config files in flask.Config.…

### LINKED ISSUE #4989: Add a file mode parameter to flask.Config.from_file()
Python 3.11 introduced native TOML support with the `tomllib` package. This could work nicely with the `flask.Config.from_file()` method as an easy way to load TOML config files:

```python
app.config.from_file("config.toml", tomllib.load)
```

However, `tomllib.load()` takes an object readable in binary mode, while `flask.Config.from_file()` opens a file in text mode, resulting in this error:

```
TypeError: File must be opened in binary mode, e.g. use `open('foo.toml', 'rb')`
```

We can get around this with a more verbose expression, like loading from a file opened with the built-in `open()` function and passing the `dict` to `app.Config.from_mapping()`:

```python
# We have to repeat the path joining that from_file() does
with open(os.path.join(app.config.root_path, "config.toml"), "rb") as file:
    app.config.from_mapping(tomllib.load(file))
```

But adding a file mode parameter to `flask.Config.from_file()` would enable the use of a simpler expression. E.g.:

```python
app.config.from_file("config.toml", tomllib.load, mode="b")
```

### PR DESCRIPTION
**Pull request by AntoineMath**
- fixes #4989 
Added optional boolean argument "text" (False by default).

### CODE REVIEW

File: src/flask/config.py [hunk 1]
```diff
@@ -245,7 +246,7 @@ def from_file(
             app.config.from_file("config.json", load=json.load)
 
             import toml
-            app.config.from_file("config.toml", load=toml.load)
+            app.config.from_file("config.toml", load=toml.load, text=True)
```
Comment (RonnyPfannschmidt) [inline src/flask/config.py:249]: this parameter is a absolute lie - when true you take away the text and drop a buffer

this should be called `binary`

and it should use the open mode

File: src/flask/config.py [hunk 2]
```diff
@@ -256,12 +257,18 @@ def from_file(
...
+            Added optional boolean argument "text" (False by default).
+            You can now use .toml config file by specifying "text=True".
+
         .. versionadded:: 2.0
         """
         filename = os.path.join(self.root_path, filename)
 
         try:
             with open(filename) as f:
+                if text:
```
Comment (davidism) [inline src/flask/config.py:270]: This should change what `open` does, not try to access `buffer` after the fact (someone else pointed out this wasn't documented as a public API).

Also, it's backwards, `text=True`, the default, should result in a text file, `False` should result in a binary file.

File: src/flask/config.py (hunk 1 above)
Comment (davidism) [reply src/flask/config.py:249]: Don't worry, I was getting there in my review.

File: src/flask/config.py [hunk 3]
```diff
@@ -256,12 +257,18 @@ def from_file(
         :param silent: Ignore the file if it doesn't exist.
         :return: ``True`` if the file was loaded successfully.
 
+        .. versionchanged:: 2.3.0
```
Comment (davidism) [inline src/flask/config.py:260]: Can drop the `.0`.

File: src/flask/config.py (hunk 1 above)
Comment (davidism) [inline src/flask/config.py:249]: This still uses the `toml` library, it needs to change to `tomllib` to be a useful example of the parameter.

File: src/flask/config.py [hunk 4]
```diff
@@ -256,12 +257,18 @@ def from_file(
         :param silent: Ignore the file if it doesn't exist.
         :return: ``True`` if the file was loaded successfully.
```
Comment (davidism) [inline src/flask/config.py:258]: Need to add documentation for the parameter.

**Review by davidism — CHANGES_REQUESTED** [review]

File: src/flask/config.py (hunk 1 above)
Comment (davidism) [reply src/flask/config.py:249]: I addressed this in my review.

**Comment by davidism** [comment]
Also, you need to fix the failing test, it should only run on Python versions that have tomllib.

**Comment by AntoineMath** [comment]
> Sorry, I think you've sort of misunderstood the feedback you got. The parameter should still be `text=True`.

I understood, but as @RonnyPfannschmidt mentioned, "binary" seems clearer. 
Because then `"text = False"` do not implicitly indicate we want to tread the file as binary

**Review by davidism — APPROVED** [review]

**Pull request merged**